### ISSUE TITLE
Epoch and Steps in Configs dont match Paper

### ISSUE THREAD

**Issue by DominikVincent** [OpenMotionLab/MotionGPT3#20]
Hello,

I noticed a discrepancy between the training details in the paper and the provided config files, and I’d like to get clarification on how you trained your model:

On the one hand, you say:

<img width="1230" height="129" alt="Image" src="https://github.com/user-attachments/assets/e4f4358a-63cd-4d19-8370-fe6a8234e0ba" />

But your config files all stages use 300 epochs.

On the other hand you say:<p>

<img width="1211" height="213" alt="Image" src="https://github.com/user-attachments/assets/51080877-fd94-4aca-8481-a83a34d5ef0b" />

<p>
that your model is trained between 160k and 300k iterations, but according to the logging, that would indicate that you train for less than 12 epochs:<p>

<img width="676" height="775" alt="Image" src="https://github.com/user-attachments/assets/1e4b3ac6-41e8-4f52-8e3b-92a5d0d83610" />

Could you clarify your training setup and update the config files to match the ones actually used for your results? Otherwise, it will be difficult for the community to fully benefit from and verify your contribution.

**Comment by BoifZ** [comment]
Thanks for the question! In our setup, each epoch contains ~800 batches. Therefore, 100 epochs = 80k iterations.

During the T2M pre-training, we train MotionGPT3 for 200 epochs (~160k iterations) and select the checkpoint with the lowest FID as the best model.

The training configuration reported in our paper reflects the evaluation setting, and the config file fields are left at their defaults. We **recommend monitoring the validation curve during your first few runs and adjusting the total training** based on the training dynamics and your specific needs.

**Comment by DominikVincent** [comment]
Thanks for the reply!

Just to double-check with you:

- The dataset contains **25,560 training samples** and **1,530 validation samples** (`val dataset loading done 1530`).
- In **stage 1**, I also observe ~800 batches/epoch with 1 GPU, and ~400 with 2 GPUs.  
- In **stage 2**, I see ~2400/1200 batches per epoch for 1/2 GPUs respectively.  

Is that aligned with your setup? Do you explicitly adjust the batch size per GPU in your configs (e.g., 2 GPUs → batch size 16 per GPU → total batch size 32)?

---

For the training, could you provide exactly the config files that you use in your final experiments, as the available config parameters vastly exceed the ones that you mention in the paper?

Based on your README and paper, is the training procedure as follows (assuming a total batch size of 32, i.e., 32 on a single GPU or 16 per GPU when using 2 GPUs)?
| Stage | Command                                                                                   | Epochs / Steps                          | Notes                                                                 |
|-------|-------------------------------------------------------------------------------------------|-----------------------------------------|-----------------------------------------------------------------------|
| 1     | `python -m train --cfg configs/MoT_vae_stage1_t2m.yaml --nodebug`                         | 200 epochs (≈160k steps, 800 batches/epoch) | All defaults in config except epochs changed to 200                            |
| 2 (part1)     | `python -m train --cfg configs/MoT_vae_stage2_all.yaml --nodebug`                         | 125 epochs (≈300k steps, 2400 batches/epoch) | All defaults in config except epochs changed to 125                            |
| 2 (part2)    | `python -m train --cfg configs/MoT_vae_stage2_instruct.yaml --nodebug`                    | ≈80k steps                              | Epochs and stage not explicitly mentioned in paper, assumed 80k iterations               |
| 3     | `python -m train --cfg configs/MoT_vae_stage3.yaml --nodebug`                             | ≈80k steps                              | Epochs not explicitly mentioned in paper, assumed 80k iterations as stated for all other training       |

Could you please confirm these settings, or share the **exact config files** you used to produce the final results in the paper, such that it is possible to recreate and verify your results?


Also, could you clarify why stage 2 is split into **two separate trainings** (multitask cross-modal learning vs. instruction finetuning)? Is this necessary?

Finally, monitoring validation is not straightforward: your code does not compute a validation loss, and the reported metrics don’t converge (see [Issue #19](https://github.com/OpenMotionLab/MotionGPT3/issues/19)). This makes it hard to follow your recommendation of “monitoring the validation curve.”

I appreciate your clarification; it will help a lot for reproducibility. Thanks!
